backend/tests/unit/agents/test_max_iterations_failure.py

```
@@ -0,0 +1,287 @@
+"""
+Test cases for MaxIterationsFailureError and stage failure detection logic.
+
+Tests the new failure detection functionality that marks stages and sessions as failed
+when max iterations is reached with the last interaction failing.
+"""
+
+import pytest
+from unittest.mock import AsyncMock, Mock, patch
```

Ruff (0.12.2)

`unittest.mock.patch` imported but unused

Remove unused import: `unittest.mock.patch`

(F401)